lib/rucio/daemons/cache/consumer.py

```diff
@@ -180,18 +116,24 @@ def run(num_thread: int = 1) -> None:
     Starts up the rucio cache consumer thread
     """
     setup_logging(process_name=DAEMON_NAME)
+    logger = formatted_logger(logging.log, DAEMON_NAME + ' %s')
 
     if rucio.db.sqla.util.is_old_db():
         raise exception.DatabaseException('Database was not updated, daemon won\'t start')
 
-    logging.info('starting consumer thread')
-    threads = [threading.Thread(target=consumer, kwargs={'id_': i, 'num_thread': num_thread})
-               for i in range(0, num_thread)]
+    logger(logging.INFO, 'starting consumer thread')
+    threads = []
+    for i in range(num_thread):
+        con_thread = threading.Thread(target=consumer, kwargs={'id_': i, 'num_thread': num_thread, 'logger': logger})
+        con_thread.start()
+        threads.append(con_thread)
 
-    [t.start() for t in threads]
-
-    logging.info('waiting for interrupts')
+    logger(logging.INFO, 'waiting for interrupts')
 
     # Interruptible joins require a timeout.
-    while threads[0].is_alive():
-        [t.join(timeout=3.14) for t in threads]
+    # while alive_threads := [thread for thread in threads if thread.is_alive()]:
+    #     for thread in alive_threads:
+    #         thread.join(timeout=3.14)
```

Comment:
Commented out code should be removed or a comment should be added to explain why it must be kept in production